src/layout/header/MenuButton.tsx

```diff
@@ -1,26 +1,36 @@
-import { Box, Button, ButtonProps } from '@mui/material';
+import { Box, Button, ButtonProps, IconButton, IconButtonProps, styled } from '@mui/material';
 import { Link, LinkProps } from 'react-router-dom';
 
-interface MenuButtonProps extends ButtonProps, Pick<LinkProps, 'to'> {
+interface StyledMenuButtonContainerProps {
   isSelected: boolean;
 }
 
-export const MenuButton = ({ isSelected, sx, ...rest }: MenuButtonProps) => (
-  <Box
-    sx={{
-      height: '100%',
-      display: 'flex',
-      borderBottom: isSelected ? '0.375rem solid white' : 'none',
-      boxShadow: isSelected ? '-1px 7px 4px -3px rgba(0,0,0,0.3)' : 'none',
-    }}>
-    <Button
-      sx={{
-        mb: isSelected ? '-0.375rem' : 'none',
-        whiteSpace: 'nowrap',
-        ...sx,
-      }}
-      LinkComponent={Link}
-      {...rest}
-    />
-  </Box>
+const StyledMenuButtonContainer = styled(Box, {
+  shouldForwardProp: (prop) => prop !== 'isSelected',
+})<StyledMenuButtonContainerProps>(({ isSelected }) => ({
+  height: '100%',
+  display: 'flex',
```

Comment:
disse to føles litt hacky, men lar det stå da det ikke er nytt og jeg ikke ser en åpenbar måte rundt det som ikke innebærer masse jobb

Reply:
ser det er mulig å flytte mye av styling fra parent Box til selve knappen, men ble da litt rot med Button kontra IconButton, så lar det være slik